lib/core/test/unit/Cardano/Wallet/Primitive/CollateralSpec.hs

```diff
@@ -0,0 +1,774 @@
+                    unit_addressSuitableForCollateral_pointerAddrGolden
+                    unit_addressSuitableForCollateral_delegationAddrGolden
+            describe "addressTypeSuitableForCollateral" $ do
+                it "satisfies same properties as addressSuitableForCollateral" $
+                    property prop_addressSuitableForCollateral_equivalence
+            describe "asCollateral" $ do
+                it "satisfies same boolean properties as addressSuitableForCollateral" $
+                    property prop_equivalence_bool
+                it "satisfies same properties as addressSuitableForCollateral" $
+                    property prop_equivalence
+
+-- The following golden keys were generated from the recovery phrase:
+-- [change twin tired knee syrup cover dog glare canvas canvas jump egg]
+
+-- cat recovery-phrase.txt | cardano-address key from-recovery-phrase Byron > root.prv
+-- cat root.prv | cardano-address key child 14H/42H | tee addr.prv | cardano-address key public --with-chain-code | cardano-address address bootstrap --root $(cat root.prv | cardano-address key public --with-chain-code) --network-tag testnet 14H/42H
+byronAddrGolden :: BL.ByteString
+byronAddrGolden = BL.fromStrict . fromJust . decodeBase58 bitcoinAlphabet $ "37btjrVyb4KFsMoVwPRZ5aJko48uBFFUnJ46eV3vC3uBCC65mj5BfbGP6jYDfhojm8MAayHo4RPvWH4x852FcJq8SHazCx31FJM2TfDpV9Azrc8UKD"
+
+-- cat recovery-phrase.txt | cardano-address key from-recovery-phrase Shelley > root.prv
+-- cat root.prv | cardano-address key child 1852H/1815H/0H/2/0 > stake.prv
+-- cat stake.prv | cardano-address key public --with-chain-code | cardano-address address stake --network-tag testnet
+stakeAddrGolden :: BL.ByteString
+stakeAddrGolden = unsafeBech32Decode
+    "stake_test1uztjkmlcknuv29pwuwd8wsk54q5eus56flqs4xy730yvnust8pvfj"
+
+-- cat recovery-phrase.txt | cardano-address key from-recovery-phrase Shelley > root.prv
+-- cat root.prv | cardano-address key child 1852H/1815H/0H/0/0 > addr.prv
+-- cat addr.prv | cardano-address key public --with-chain-code | cardano-address address payment --network-tag 0 | cardano-address address pointer 42 14 0
+pointerAddrGolden :: BL.ByteString
+pointerAddrGolden  = unsafeBech32Decode
+    "addr_test1gpdylg53ekxh2404mfgw4pt4gfm7dc9dkc74ck0gtrld8up2pcqqefucl2"
+
+-- cat recovery-phrase.txt | cardano-address key from-recovery-phrase Shelley > root.prv
+-- cat root.prv | cardano-address key child 1852H/1815H/0H/2/0 > stake.prv
+-- cat root.prv | cardano-address key child 1852H/1815H/0H/0/0 > addr.prv
+-- cat addr.prv | cardano-address key public --with-chain-code | cardano-address address payment --network-tag testnet | cardano-address address delegation $(cat stake.prv | cardano-address key public --with-chain-code)
+delegationAddrGolden :: BL.ByteString
+delegationAddrGolden = unsafeBech32Decode "addr_test1qpdylg53ekxh2404mfgw4pt4gfm7dc9dkc74ck0gtrld8uyh9dhl3d8cc52zacu6wapdf2pfnepf5n7pp2vfaz7ge8eqd4nn9s"
```

Comment:
This might make it easier to read in an editor:
```suggestion
delegationAddrGolden = unsafeBech32Decode 
    "addr_test1qpdylg53ekxh2404mfgw4pt4gfm7dc9dkc74ck0gtrld8uyh9dhl3d8cc52zacu6wapdf2pfnepf5n7pp2vfaz7ge8eqd4nn9s"
```